packages/next/src/next-devtools/dev-overlay/components/devtools-panel/devtools-panel.tsx

```diff
@@ -39,11 +40,16 @@ export function DevToolsPanel({
   getSquashedHydrationErrorDetails: (error: Error) => HydrationErrorState | null
 }) {
   const [activeTab, setActiveTab] = useState<DevToolsPanelTabType>('issues')
-  const [isFullscreen, setIsFullscreen] = useState(false)
+  const [isFullscreen, setIsFullscreen] = useState(state.isErrorOverlayOpen)
   const [vertical, horizontal] = state.devToolsPosition.split('-', 2)
 
+  useEffect(() => {
+    setIsFullscreen(state.isErrorOverlayOpen)
+  }, [state.isErrorOverlayOpen])
```

Comment:
[You don't need useEffect](https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes) here:
```suggestion
  if (isFullscreen !== state.isErrorOverlayOpen) {
    setIsFullscreen(state.isErrorOverlayOpen)
  }
```

But also: why do we need two separate states at all?